DBMS/Assignment 3/DBMSAssignment 3.sql

```diff
@@ -0,0 +1,171 @@
+INSERT INTO publishers VALUES(4,'d');
+INSERT INTO publishers VALUES(5,'e');
+
+INSERT INTO members VALUES (1000,"udit","abc","xyz","f");
+INSERT INTO members(member_name,addressline1,addressline2,category) VALUES('nagar','cab','xyz','s');
+INSERT INTO members(member_name,addressline1,addressline2,category) VALUES('udit','abc','xyz','s');
+INSERT INTO members(member_name,addressline1,addressline2,category) VALUES('udit','abc','xyz','f');
+INSERT INTO members(member_name,addressline1,addressline2,category) VALUES('udit','abc','xyz','o');
+
+
+INSERT INTO subjects VALUES(700,"dbms");
+INSERT INTO subjects VALUES(701,"adbms");
+INSERT INTO subjects VALUES(702,"dsa");
+INSERT INTO subjects VALUES(703,"pf");
+INSERT INTO subjects VALUES(704,"java");
+
+INSERT INTO title VALUES(1200,"abc",700,1);
+INSERT INTO title VALUES(1201,"ab",700,1);
+INSERT INTO title VALUES(1202,"a",701,2);
+INSERT INTO title VALUES(1203,"b",701,2);
+INSERT INTO title VALUES(1204,"c",704,3);
+
+INSERT INTO books(accession_no,title_id,purchase_date,price,status) VALUES(505,1200,"2015-05-21",100.00,1);
+INSERT INTO books(accession_no,title_id,purchase_date,price,status) VALUES(506,1201,"2015-05-21",100.00,1);
+INSERT INTO books(accession_no,title_id,purchase_date,price,status) VALUES(507,1201,"2015-05-21",120.00,1);
+INSERT INTO books(accession_no,title_id,purchase_date,price,status) VALUES(508,1202,"2015-05-21",130.00,1);
+INSERT INTO books(accession_no,title_id,purchase_date,price,status) VALUES(509,1200,"2015-05-21",140.00,1);
+
+INSERT INTO title_author VALUES(1200,102);
+INSERT INTO title_author VALUES(1201,102);
+INSERT INTO title_author VALUES(1202,101);
+INSERT INTO title_author VALUES(1202,104);
+INSERT INTO title_author VALUES(1204,104);
+
+INSERT INTO book_issue VALUES("2015-01-04",505,1000,"2015-02-02");
+INSERT INTO book_issue VALUES("2015-01-03",505,1000,"2015-02-02");
+INSERT INTO book_issue VALUES("2015-01-02",506,1001,"2015-04-02");
+INSERT INTO book_issue VALUES("2015-01-01",506,1001,"2015-04-02");
+INSERT INTO book_issue VALUES("2015-01-04",506,1001,"2015-01-15");
+
+INSERT INTO book_return VALUES("2015-01-04",505,1000,"2015-01-02");
+INSERT INTO book_return VALUES("2015-03-03",505,1000,"2014-01-02");
+INSERT INTO book_return VALUES("2015-02-02",506,1001,"2014-01-02");
+INSERT INTO book_return VALUES("2015-05-01",507,1001,"2011-01-02");
+INSERT INTO book_return VALUES("2015-10-04",507,1002,"2014-01-02");
+
+-- ASSIGNMENT 1
+-- QN 1
+SELECT m.member_name,m.member_Id,t.title_Name,t.title_Id,b.accession_No,b.issue_Date,b.due_Date , DATEDIFF(due_Date,issue_Date ) AS DiffDate  from Members m,Book_Issue b,Title t, Books bk
+WHERE (m.member_id = b.member_id && t.title_id = bk.title_id && bk.accession_No = b.accession_No) &&( DATEDIFF(b.due_Date,b.issue_Date )) > 60 order By member_name ,title_Name;
+
+-- QN 2
+
+Select member_name,length(member_name) as lengthOfName from members where length(member_name) in (Select max(length(member_name)) from members);
```

Comment:
ACTION: there will be only one max So use = instead of In, if you are giving a name to the length then use that name in where condition instead of length(member_name)